IE-413 Configure for Ruby 3

ncperry opened the pull request:
Fix for the error: 

> System error from Qualtrics Responses - Check
> for Risk Assessment Scorecard. no implicit
> conversion of Faraday::Utils::Headers into String

We had the same issue that we've seen elsewhere; keyword arguments were getting forwarded improperly.

See the [Ruby Docs](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/) for context.

ncperry committed:
chore: configure for ruby 3

ncperry committed:
fix: conform to ruby 3 keyword argument format

wilkinsondesouza commented on qualtrics_api.gemspec:
```diff
@@ -20,9 +20,9 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
-  spec.required_ruby_version = '>= 2.3.0'
+  spec.required_ruby_version = '>= 3.0.6'
```
💭  Does the new code run in Ruby 2.x? 

- If they do I don't feel that we need to change this requirement. 
- If they don't, it's for the best that we bump the gem version as well

ncperry replied:
The new test will not pass in Ruby 2.x. Since we're the only consumers of this gem, it made sense to me to bump the ruby version rather than write a more complex test. I think it's a good idea to bump the gem version as well.

wilkinsondesouza replied:
LG!

ncperry committed:
chore: update gem version number

ncperry committed:
fix: update to ruby 3 keyword argument syntax

b6b reviewed (approved):
LGTM!

wilkinsondesouza reviewed (approved):
LGTM!

ncperry committed:
chore: run rspec in github actions

ncperry committed:
fix: indentation

ncperry committed:
use old bundler version

ncperry committed:
Update github action step name

ncperry committed:
fix: update to ruby 3 keywork arguments

ncperry committed:
fix: update to ruby 3 keyword format

ncperry committed:
fix: update to ruby 3 keyword format

b6b committed:
Fix nested JSON key case

In https://github.com/mavenlink/qualtrics_api/pull/17 we changed response parsing to no longer deep-snake-case nested JSON. It seems like we forgot to update these tests to match the new behavior.

Co-authored-by: Nate Perry <nperry@mavenlink.com>
